### ISSUE TITLE
Error on running the plugin in executable jar

### ISSUE THREAD

**Issue by dikey94** [dadoonet/elasticsearch-beyonder#32]
Hi,
thanks for the Beyonder - I think it will be useful.

I'm working with Spring Boot application and I'm using Maven for building process. 

The plugin works fine locally and everything works fine on my local machine during testing.

The problem appears on prod, where the app runs as executable jar. Running

> ElasticsearchBeyonder.start(restClient);

does not do... anything. I've been investigating the situation for a few hours and realized that no .json file is founded.

I've run a part of your code locally. In ResourceList class at line 90 you check

> name.startsWith(root)

but in my case this condition is never true. I have to change my command to
> ElasticsearchBeyonder.start(restClient, "BOOT-INF/classes/elasticsearch");

to start working.

Not sure if it's a bug or on issue with my .jar.

I'd appreciate any feedback. Thanks.

**Comment by dikey94** [comment]
up

**Comment by dadoonet** [comment]
Sorry for the delay. I thought you were able to solve it by yourself by using:

```java
ElasticsearchBeyonder.start(restClient, "BOOT-INF/classes/elasticsearch");
```

Did I misunderstand?

If not, I feel like I "just" need to document it?

**Comment by dikey94** [comment]
Thanks.
Not sure if it's only has to be documented - I treat it as a "dirty" solution. 

```
public void loadIndices() throws Exception {
        if (environment.acceptsProfiles("prod")) {
            ElasticsearchBeyonder.start(restClient, "BOOT-INF/classes/elasticsearch");
        } else {
            ElasticsearchBeyonder.start(restClient);
        }
}
```

**Comment by dadoonet** [comment]
Do you have a sample basic project I could test that from to see if a patch can actually fix it?

**Comment by romankh** [comment]
Hi, 

while I dont have a sample project, I can confirm the issue, as I am having the same.
When developing in the IDE, it works without any problem, while when running the built jar with "java -jar myapp.jar" I have the same issue, the index is not initialized as the IndexFinder.java does not find any files.

I had a similar issues when accessing resources that were in the jar file but where accessed as file, I believe the solution was to load the resource as stream instead as file.

**Comment by rudibatt** [comment]
We have the same issue...
What about a solution that extracts the path prefix from the dirURL, that was already found correctly? 
The following code works for us, but I haven't found the time to write tests that cover also other similar usecases.
https://github.com/CommerceExperts/elasticsearch-beyonder/commit/540a1f9938bd6a5f883f220ce8f3c5e35dcfd02b

**Comment by romankh** [comment]
@dadoonet,@rudibatt: would you provide / accept a pull request for this issue
If necessary, I can add the some testing

**Comment by dadoonet** [comment]
Closed by #43 thanks to @rudibatt!
